### PR TITLE
dev-libs/libbpf: add live ebuild

### PR DESCRIPTION
**Pull request by jkelecic**
Signed-off-by: Josip Kelecic <josip.kelecic@sartura.hr>

### CODE REVIEW

File: dev-libs/libbpf/libbpf-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs git-r3
+
+DESCRIPTION="Stand-alone build of libbpf from the Linux kernel"
```
Comment (charIes17) [inline dev-libs/libbpf/libbpf-9999.ebuild:8]: What about HOMEPAGE?
HOMEPAGE="https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git"

Comment (jkelecic) [reply dev-libs/libbpf/libbpf-9999.ebuild:9]: Homepage added, thank you for advice.

File: dev-libs/libbpf/libbpf-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+EAPI=7
+
+inherit toolchain-funcs git-r3
+
+HOMEPAGE="https://www.kernel.org/doc/html/latest/bpf/bpf_devel_QA.html"
+DESCRIPTION="Stand-alone build of libbpf from the Linux kernel"
+EGIT_REPO_URI="https://github.com/libbpf/libbpf.git"
+
+
```
Comment (chutz) [inline dev-libs/libbpf/libbpf-9999.ebuild:12]: 2 blank lines in a row are forbidden in `::gentoo`

As a side note, changes do not need to be added as a separate PR, you can update the commit with `git commit --amend` then do a `git push --force` over the previous branch. Force pushing to your own fork is not rude or forbidden.

File: dev-libs/libbpf/libbpf-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs git-r3
```
Comment (chutz) [inline dev-libs/libbpf/libbpf-9999.ebuild:6]: Generally in packages with a live ebuild, for ease of maintenance I prefer to do something like:
```ebuild
inherit toolchain-funcs
if [[ ${PV} =~ [9]{4,} ]]; then
	inherit git-r3
	EGIT_REPO_URI="https://github.com/libbpf/libbpf.git"
else
	SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
	KEYWORDS="~amd64 ~arm64 ~x86"
fi
```
That way the 9999 can often be identical to the latest version of the ebuild, barring patches.

File: dev-libs/libbpf/libbpf-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+HOMEPAGE="https://www.kernel.org/doc/html/latest/bpf/bpf_devel_QA.html"
+DESCRIPTION="Stand-alone build of libbpf from the Linux kernel"
+EGIT_REPO_URI="https://github.com/libbpf/libbpf.git"
+
+
+LICENSE="GPL-2 LGPL-2.1 BSD-2"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm64 ~x86"
```
Comment (chutz) [inline dev-libs/libbpf/libbpf-9999.ebuild:15]: Live ebuilds should *never* have keywords set. This line should be removed.

**Comment by jkelecic** [comment]
@chutz Thank you for reviewing, changes have been made accordingly to your advice.
